### PR TITLE
docs(bazel): added quickstart for working with bazel

### PR DESCRIPTION
**Pull request by ducin**
The quickstart includes most important information to contributors with no experience with Bazel. THe aim of the document is to help troubleshooting the most common scenarios.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: docs/BAZEL_QUICKSTART.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        include = ["*.ts"],
+        exclude = ["*.spec.ts"],
+    ),
+    deps = [
+        "//devtools/projects/ng-devtools-backend/src/lib/directive-forest",
+        "//devtools/projects/protocol",
+    ],
+)
+```
+doesn't include the `@angular/code` in its `deps`. Add the `"//packages/core",`:
```
Comment (eneajaho) [inline docs/BAZEL_QUICKSTART.md:110]:
```suggestion
doesn't include the `@angular/core` in its `deps`. Add the `"//packages/core",`:
```

Comment (ducin) [reply docs/BAZEL_QUICKSTART.md]: @eneajaho done! thx

**Comment by ducin** [comment]
@josephperrott would you find 5 minutes to take a look?

**Comment by bencodezen** [comment]
This is a great start @ducin! Appreciate you taking the time to write this. I can't speak to the technical accuracy of the document, but this is a great foundation for us to work with going forward.

Nice work! 🙌

**Review by bencodezen — APPROVED** [review]

File: docs/BAZEL_QUICKSTART.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
+# Bazel Quickstart
+
+This document is intended for Angular contributors who have no (or hardly any) experience with [Bazel](./BAZEL.md). It includes some common troubleshooting examples which will help you get started efficiently.
+
+## Overview
+
+Although Bazel itself is written in various programming languages, what we can see in the Angular repo is actually Python declarations. The example [`devtools/tools/ng_module.bzl`](https://github.com/angular/angular/blob/c62b2dae8c4653996969df992e8f88887e7c83c4/devtools/tools/ng_module.bzl) file includes a declaration of `ng_module`:
```
Comment (alxhub) [inline docs/BAZEL_QUICKSTART.md:7]: Nit: Bazel uses a language called Starlark, which is a variant of Python.

**Comment by AndrewKushnir** [comment]
(removing the "merge" label, we still need a review from @josephperrott or @devversion)

File: docs/BAZEL_QUICKSTART.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        "//packages/forms",
+        "//packages/platform-browser",
+        "//packages/platform-browser-dynamic",
+        "//packages/router",
+        "//packages/zone.js/lib",
+...
+```
+in Bazel config files (`packages/*` points to `packages/*`, and so `packages/core` points to `@angular/core`).
+
+## Barrel files
```
Comment (josephperrott) [inline docs/BAZEL_QUICKSTART.md:271]: I would not include this section as barrel files are not a bazel specific thing, the decision to have barrel files in the javascript ecosystem is managed separately from bazel.

**Comment by JeanMeche** [comment]
@ducin Do you wish to take into account the comments ?

**Comment by thePunderWoman** [comment]
Closing since there's been no activity on this. Feel free to open a new PR if you'd like to see this change land.

**Pull request closed without merging**